File: 백준/Silver/10815. 숫자 카드/숫자 카드.py
import sys
input = sys.stdin.readline
n = int(input())
cards = list(map(int, input().split())) #상근 카드
m = int(input())
cards2 = list(map(int, input().split())) #상근..확인차..
res = []
card_set = set(cards)
for card in cards2:
    if card in card_set:
        res.append(1)
    else:
        res.append(0)
print(*res)

File: 백준/Silver/10815. 숫자 카드/숫자 카드.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Set<Integer> cardSet = new HashSet<>();
        for(int i=0;i<n;i++){
            cardSet.add(sc.nextInt());
        }
        int m = sc.nextInt();
        int[]res = new int[m];
        for(int i=0;i<m;i++){
            int card = sc.nextInt();
            if(cardSet.contains(card)){
                res[i] = 1;
            }else{
                res[i]= 0;
            }
        }
        for(int result : res){
            System.out.print(result + " ");
        }
        
    }
}